### DISCUSSION TITLE
Error: The module or file Module*Query_graphql can't be found.

### DISCUSSION THREAD

**Discussion by MoOx** [zth/reason-relay#335] [Q&A] [answered]
Hi,

I am starting to use ReScriptRelay.
I read [the doc](https://rescript-relay-documentation.vercel.app/docs/making-queries) properly about how to make query but I am can't have a thing to compile.

Here is a basic example of what I am doing:

```res
/* src/pages/PageAuthSignup.res */
module Query = %relay(`
  query PageAuthSignupUserQuery {
    users {
      id
      email
    }
  }
`)
```

And the error I got is 

<img width="685" alt="image" src="https://user-images.githubusercontent.com/157534/164993697-dba972c5-44c6-40e2-b425-9f8fa33ad6f3.png">

```
The module or file PageAuthSignupUserQuery_graphql can't be found.
- If it's a third-party dependency:
  - Did you list it in bsconfig.json?
  - Did you run `rescript build` instead of `rescript build -with-deps`
    (latter builds third-parties)?
- Did you include the file's directory in bsconfig.json?
```

I respected the naming convention so I don't get what means the error.

FYI, I am using Vscode with appropriate plugins, Next.js, Relay, ReScript (obviously) and have properly configured relay with the schema retrieved using `get-graphql-schema` (autocompletion worked for `user<tab>` to give me proper solutions, so it seems to be configured well enough (also note on the screenshot the type at the top, properly discovered `directus_users.email: String`).

Any help is appreciated !

**Comment by MoOx** [comment, marked as answer]
Using `rescript-relay-compiler` CLI helped me to discover new errors that didn't show up in Vscode. In my case: I didn't created `artifactDirectory`.
Creating the folder helped to have code generated properly.

**Reply by zth** [reply]
Glad you figured it out, sorry I missed this! 

The compiler still has a few rough edges since the transition from JS to Rust, and this is one of them. I believe it used to automatically create the directory if it didn't exist.

I'll make a note about adding this to the docs. Might even be a good idea to have a proper `init` command in the `rescript-relay-cli` that takes care of all of this. I have that in the RescriptRelay router project I'm doing, and it does save quite a few frustrating issues like this.
